src/AWS.Deploy.Orchestration/Utilities/TemplateMetadataReader.cs

```diff
@@ -45,14 +45,45 @@ public async Task<CloudApplicationMetadata> LoadCloudApplicationMetadata(string
 
             var response = await client.GetTemplateAsync(request);
 
-            return ReadSettings(response.TemplateBody);
+            if(IsJsonCFTemplate(response.TemplateBody))
+                return ReadSettingsFromJSONCFTemplate(response.TemplateBody);
+            else
+                return ReadSettingsFromYAMLCFTemplate(response.TemplateBody);
         }
 
         /// <summary>
-        /// Read the AWS .NET deployment tool metadata from the CloudFormation template.
+        /// Read the AWS .NET deployment tool metadata from the CloudFormation template which is in JSON format.
         /// </summary>
         /// <returns></returns>
-        private static CloudApplicationMetadata ReadSettings(string templateBody)
+        private static CloudApplicationMetadata ReadSettingsFromJSONCFTemplate(string templateBody)
+        {
+            try
+            {
+                var cfTemplate = JsonConvert.DeserializeObject<CFTemplate>(templateBody);
+
+                var cloudApplicationMetadata = new CloudApplicationMetadata(
+                    cfTemplate?.Metadata?[Constants.CloudFormationIdentifier.STACK_METADATA_RECIPE_ID] ??
+                        throw new Exception("Error parsing existing application's metadata to retrieve Recipe ID."),
+                    cfTemplate?.Metadata?[Constants.CloudFormationIdentifier.STACK_METADATA_RECIPE_VERSION] ??
+                        throw new Exception("Error parsing existing application's metadata to retrieve Recipe Version.")
```

Comment:
We should be throwing a custom exception with an error code.

Reply:
Be sure the `Exception`s thrown in `ReadSettingsFromYAMLCFTemplate` also have error codes. Should they be the same error code or is it useful to have different error codes for JSON vs YAML?

Reply:
These exceptions are instantly caught by a catch block which throws a custom exception with an error code. I don't think we care to make a distinction between an invalid recipe id or recipe version metadata. Anything corrupted in the metadata should be treated the same in my option.

Reply:
Since ReadSettingsFromYAMLCFTemplate and ReadSettingsFromJSONCFTemplate are basically doing the same thing with the difference of the type, I think they should be treated the same. A problem in the metadata is a problem in the metadata, whether the metadata is YAML or JSON.